shared/utility/actuation/InverseKinematics.hpp

```diff
@@ -42,177 +42,143 @@ namespace utility::actuation::kinematics {
-            (isAnkleAboveWaist ? -1 : 1) * (Eigen::Matrix<Scalar, 3, 1>::UnitZ() - (cosZandHipX * hipX));
-        const Scalar legPlaneGlobalZLength = legPlaneGlobalZ.norm();
-        if (legPlaneGlobalZLength > 0) {
-            legPlaneGlobalZ /= legPlaneGlobalZLength;
-        }
-
-        const Scalar cosHipRoll = legPlaneGlobalZ.dot(Eigen::Matrix<Scalar, 3, 1>::UnitZ());
-        hipRoll = (hipRollPositive ? 1 : -1) * std::acos(std::max(std::min(cosHipRoll, Scalar(1)), Scalar(-1)));
-
-        const Scalar phi4 = M_PI - knee - lowerLeg;
-        // Superposition values:
-        const Scalar sinPIminusPhi2 = std::sin(M_PI - phi2);
-        const Eigen::Matrix<Scalar, 3, 1> unitUpperLeg =
-            unitTargetLeg * (std::sin(phi2 - phi4) / sinPIminusPhi2) + ankleY * (std::sin(phi4) / sinPIminusPhi2);
-        const bool isHipPitchPositive = hipX.dot(unitUpperLeg.cross(legPlaneGlobalZ)) >= 0;
-
-        hipPitch = (isHipPitchPositive ? 1 : -1)
-                   * std::acos(std::max(std::min(legPlaneGlobalZ.dot(unitUpperLeg), Scalar(1)), Scalar(-1)));
-
-        // If leg is above waist then hipX is pointing in the wrong direction in the xy plane
-        Eigen::Matrix<Scalar, 3, 1> hipXProjected = (isAnkleAboveWaist ? -1 : 1) * hipX;
-        hipXProjected.z()                         = 0;
-        hipXProjected.normalize();
-        const bool isHipYawPositive = hipXProjected.dot(Eigen::Matrix<Scalar, 3, 1>::UnitY()) >= 0;
-        hipYaw                      = (isHipYawPositive ? 1 : -1)
-                 * std::acos(std::max(std::min(hipXProjected.dot(Eigen::Matrix<Scalar, 3, 1>::UnitX()), Scalar(1)),
-                                      Scalar(-1)));
+        const Scalar cos_hip_roll = leg_plane_global_z.dot(Eigen::Matrix<Scalar, 3, 1>::UnitZ());
+        const Scalar hip_roll =
+            (hip_roll_positive ? 1 : -1) * std::acos(std::max(std::min(cos_hip_roll, Scalar(1)), Scalar(-1)));
+
+        // Hip pitch
+        const Scalar phi4               = knee_pitch - lower_leg;
+        const Scalar sin_pi_minus_phi_2 = std::sin(M_PI - phi2);
+        const Eigen::Matrix<Scalar, 3, 1> unit_upper_leg =
+            unit_target_leg * (std::sin(phi2 - phi4) / sin_pi_minus_phi_2)
+            + ankle_y * (std::sin(phi4) / sin_pi_minus_phi_2);
+        const bool is_hip_pitch_positive = hip_x.dot(unit_upper_leg.cross(leg_plane_global_z)) >= 0;
+        const Scalar hip_pitch =
+            (is_hip_pitch_positive ? -1 : 1)
+            * std::acos(std::max(std::min(leg_plane_global_z.dot(unit_upper_leg), Scalar(1)), Scalar(-1)));
+
+        // Hip yaw
+        // If leg is above waist then hip_x is pointing in the wrong direction in the xy plane
+        Eigen::Matrix<Scalar, 3, 1> hip_x_projected = (is_ankle_above_waist ? -1 : 1) * hip_x;
+        hip_x_projected.z()                         = 0;
+        hip_x_projected.normalize();
+        const bool is_hip_yaw_positive = hip_x_projected.dot(Eigen::Matrix<Scalar, 3, 1>::UnitY()) >= 0;
```

Comment:
Dotting with unit Y is just the same as accessing the y component
```suggestion
        const bool is_hip_yaw_positive = hip_x_projected.y() >= 0;
```